### PR TITLE
Stabilise upper bound of `SGPR`.

### PR DESCRIPTION
**Pull request by markvdw**
Improved numerical stability of upper bound.

### CODE REVIEW

File: gpflow/models/sgpr.py [hunk 1]
```diff
@@ -59,10 +59,12 @@ def upper_bound(self):
         kuu = Kuu(self.inducing_variable, self.kernel, jitter=default_jitter())
         kuf = Kuf(self.inducing_variable, self.kernel, x_data)
 
-        KufKfu = tf.linalg.matmul(kuf, kuf, transpose_b=True)
+        I = tf.eye(tf.shape(kuu)[0], dtype=default_float())
 
         L = tf.linalg.cholesky(kuu)
-        LB = tf.linalg.cholesky(kuu + KufKfu / self.likelihood.variance)
+        A = tf.linalg.triangular_solve(L, kuf, lower=True)
+        B = I + tf.linalg.matmul(A, A, transpose_b=True) / self.likelihood.variance
```
Comment (awav) [inline gpflow/models/sgpr.py]: You can compute it only once:  tf.linalg.matmul(A, A, transpose_b=True)

Comment (markvdw) [reply gpflow/models/sgpr.py]: Will do.

File: doc/source/notebooks/theory/upper_bound.ipynb [hunk 2]
```diff
@@ -322,8 +331,8 @@
      "name": "stdout",
      "output_type": "stream",
      "text": [
-      "Lower bound: -65.059355\n",
-      "Upper bound: -65.058996\n"
+      "Lower bound: -62.487148\n",
+      "Upper bound: -62.479897\n"
```
Comment (st--) [inline doc/source/notebooks/theory/upper_bound.ipynb]: do we know why it changed ? are we not setting the random seed? would be good to have notebooks be reproducible ...

File: gpflow/models/sgpr.py [hunk 3]
```diff
@@ -49,6 +49,11 @@ class SGPRUpperMixin(GPModel):
         year={2014},
         month={Dec}
       }
+
+    The key quantity, the trace term, can be computed with
+            # To compute each individual element of the trace term, use...
```
Comment (st--) [inline gpflow/models/sgpr.py]: do we need both of these lines? seems redundant

**Pull request merged**